app/gui2/shared/retry.ts

```diff
@@ -18,6 +18,9 @@ export interface BackoffOptions<E> {
   ) => boolean | void
   /** Called right before returning. */
   onSuccess?: (retryCount: number) => void
+  /** Called right before throwing an error. Note that `onBeforeRetry` is called for all tries
+   * before the last one. */
```

Comment:
What does it mean, "all tries before last one"? From the name, I'd rather assume that the last one will also have called `onBeforeRetry` before.

Reply:
i guess the docs are confusing. this is called on the final retry, when there is no next retry (and so there is no delay). alternatively we can simply call `onBeforeRetry` with a `delay === null` and modify the interface as appropriate